doc/examples/qsv_transcode.c

```diff
@@ -0,0 +1,442 @@
+        break;
+    default:
+        fprintf(stderr, "Codec is not supportted by qsv\n");
+        return AVERROR(ENAVAIL);
+    }
+
+    if (!(decoder_ctx = avcodec_alloc_context3(decoder)))
+        return AVERROR(ENOMEM);
+
+    if ((ret = avcodec_parameters_to_context(decoder_ctx, video->codecpar)) < 0) {
+        fprintf(stderr, "avcodec_parameters_to_context error. Error code: %s\n",
+                av_err2str(ret));
+        return ret;
+    }
+    decoder_ctx->framerate = av_guess_frame_rate(ifmt_ctx, video, NULL);
+
+    decoder_ctx->hw_device_ctx = av_buffer_ref(hw_device_ctx);
+    if (!decoder_ctx->hw_device_ctx) {
+        fprintf(stderr, "A hardware device reference create failed.\n");
+        return AVERROR(ENOMEM);
+    }
+    decoder_ctx->get_format    = get_format;
+    decoder_ctx->pkt_timebase = video->time_base;
+    if ((ret = avcodec_open2(decoder_ctx, decoder, NULL)) < 0)
+        fprintf(stderr, "Failed to open codec for decoding. Error code: %s\n",
+                av_err2str(ret));
+
+    return ret;
+}
+
+static int encode_write(AVPacket *enc_pkt, AVFrame *frame)
+{
+    int ret = 0;
+
+    av_packet_unref(enc_pkt);
+
+    if((ret = dynamic_set_parameter(encoder_ctx)) < 0) {
+        fprintf(stderr, "Failed to set dynamic parameter. Error code: %s\n",
+                av_err2str(ret));
+        goto end;
+    }
+
+    if ((ret = avcodec_send_frame(encoder_ctx, frame)) < 0) {
+        fprintf(stderr, "Error during encoding. Error code: %s\n", av_err2str(ret));
+        goto end;
+    }
+    while (1) {
+        if (ret = avcodec_receive_packet(encoder_ctx, enc_pkt))
+            break;
+        enc_pkt->stream_index = 0;
+        av_packet_rescale_ts(enc_pkt, encoder_ctx->time_base,
+                             ofmt_ctx->streams[0]->time_base);
+        if ((ret = av_interleaved_write_frame(ofmt_ctx, enc_pkt)) < 0) {
+            fprintf(stderr, "Error during writing data to output file. "
+                    "Error code: %s\n", av_err2str(ret));
+            return ret;
+        }
+    }
+
+end:
+    if (ret == AVERROR_EOF)
+        return 0;
+    ret = ((ret == AVERROR(EAGAIN)) ? 0:-1);
+    return ret;
+}
+
+static int dec_enc(AVPacket *pkt, const AVCodec *enc_codec, char *optstr)
+{
+    AVFrame *frame;
+    int ret = 0;
+
+    ret = avcodec_send_packet(decoder_ctx, pkt);
+    if (ret < 0) {
+        fprintf(stderr, "Error during decoding. Error code: %s\n", av_err2str(ret));
+        return ret;
+    }
+
+    while (ret >= 0) {
+        if (!(frame = av_frame_alloc()))
+            return AVERROR(ENOMEM);
+
+        ret = avcodec_receive_frame(decoder_ctx, frame);
+        if (ret == AVERROR(EAGAIN) || ret == AVERROR_EOF) {
+            av_frame_free(&frame);
+            return 0;
+        } else if (ret < 0) {
+            fprintf(stderr, "Error while decoding. Error code: %s\n", av_err2str(ret));
+            goto fail;
+        }
+        if (!initialized) {
```

Comment:
Use the check below instead ? If so, `initialized` is not needed. 
`if (!encoder_ctx->hw_frames_ctx) {`

Reply:
Good advice. Done. Thanks